src/maps/OverlayedMap.cpp

```diff
@@ -393,12 +410,26 @@ void OverlayedMap::drawAirportICAORing(const xdata::Airport& airport, int x, int
     }
 }
 
-void OverlayedMap::drawAirportText(const xdata::Airport& airport, int x, int y, uint32_t color) {
-    int zoomLevel = stitcher->getZoomLevel();
-    int idSize = zoomLevel * 2;
-    mapImage->drawText(airport.getID(), idSize, x, y + 20, color);
-    if (zoomLevel >= 11) {
-        mapImage->drawText(airport.getName(), idSize * 0.7, x, y + 20 + idSize, color);
+void OverlayedMap::drawAirportText(const xdata::Airport& airport, int x, int y, double mapWidthNM, uint32_t color) {
+    // Place text below southern airport boundary and below symbol
+    int yOffset = y + 15;
+    auto &locDownRight = airport.getLocationDownRight();
+    if (locDownRight.isValid()) {
+        int xIgnored;
+        positionToPixel(locDownRight.latitude, locDownRight.longitude, xIgnored, yOffset);
+        yOffset += 15;
+    }
+    if (mapWidthNM > 40) {
+        mapImage->drawText(airport.getID(), 16, x, yOffset, color, img::COLOR_WHITE & 0xA0FFFFFF, img::Align::CENTRE);
+    } else {
+        std::string nameAndID = airport.getName() + " (" + airport.getID() + ")";
+        std::string elevationFeet = std::to_string(airport.getElevation());
+        int rwyLengthHundredsFeet = (int)((airport.getLongestRunwayLength() * 3.28) / 100.0);

```

Comment:
The cast to int is not required since the left of the expression is an int anyways. Removing the cast makes it more readable in my opinion.